### DISCUSSION TITLE
希望增加到PANTHERBAR  extensions

### DISCUSSION THREAD

**Discussion by zhxiyao** [crimx/ext-saladict#605] [Show and tell]
RT，Windows 下的 PopClip， 貌似这个软件还比较新鲜，上面能用的插件还比较少，但的确和mac下的POPCLIP神似，希望SaladDict可以通过PantherBar呼出，Quicher需要点击中键且出来的菜单也蛮大的，体验还是不如PopClip。
https://www.microsoft.com/zh-cn/p/pantherbar/9npz2tvkjvt7?rtc=1&activetab=pivot:overviewtab
https://pantherbar-app.com/

**Comment by crimx** [comment]
原理都是一样的（剪贴板 + Saladict 全局快捷键），你可以自行配置。

**Comment by hqyyqh** [comment]
这个也是可以的，使用[文档](http://pantherbar-app.com/create-extension)中的 `keyCombo` 选项或者在 `PowerShell` 脚本中调用快捷键就可以。

> Pantherbar 大部分场合可用，在少数网页跟软件内不可用（例如 Acrobat）

## 效果展示

### 划词操作
![PantherbarSaladict](https://user-images.githubusercontent.com/38676455/71808606-2f561b80-30a9-11ea-89ca-1ceb128d751f.gif)

### 分段识别
![Segment](https://user-images.githubusercontent.com/38676455/71808629-3e3cce00-30a9-11ea-8941-fb8914396403.jpg)

## PantherBar 沙拉查词扩展下载

快捷键设置项【在独立窗口中搜索剪贴板内容】为 Alt+L，全局（沙拉查词快捷键具体设置参考 #493 ）。下载文件后缀`.zip` 改为 `.pbar`，双击安装。

**自定义快捷键：** 如需自定义调用快捷键，下载压缩包后修改 `.ps1` 文件内的 `$ShortCut` 变量，重新压缩修改后缀即可。

> [SaladictTranslate.v1.9.zip](https://github.com/user-attachments/files/22869098/SaladictTranslate.v1.9.zip)
> fix: 修复出现小键盘锁定的提示（Num Lock）
> 感谢 @BenmiaoZ 提供的[解决方案](https://github.com/crimx/ext-saladict/discussions/605#discussioncomment-14655211)

### 历史版本
--------------
> [SaladictTranslate v1.8.zip](https://github.com/crimx/ext-saladict/files/4022902/SaladictTranslate.v1.8.zip)
>fix: 删除空行
>add: 标题判断
--------------
>[SaladictTranslate v1.6.zip](https://github.com/crimx/ext-saladict/files/4010405/SaladictTranslate.v1.6.zip)
>fix: 误判行末空白字符为分段
-------------
> [SaladictTranslate v1.4.zip](https://github.com/crimx/ext-saladict/files/4007678/SaladictTranslate.v1.4.zip)
>1. 解决 Powershell 输出到剪贴板的编码问题
>2. 完善识别分段功能，基本能够还原 PDF 的分段格式
>3. 替换 LaTeX 生成 PDF 中 `ﬂ ﬃ ﬁ ﬀ` 字符
>**（插件基本完美可用了）**
>![Snipaste_2019-12-28_22-21-53_compress](https://user-images.githubusercontent.com/38676455/71545435-f097d580-29c5-11ea-92e4-2280511ec776.jpg)
-----------------------------
>[SaladictTranslate v1.2.zip](https://github.com/crimx/ext-saladict/files/4006901/SaladictTranslate.v1.2.zip)
>增加标点识别分段
----------------------
>[SaladictTranslate.zip v1.1](https://github.com/crimx/ext-saladict/files/4006673/SaladictTranslate.zip)
>增加是否删除分段选项
>![text](https://user-images.githubusercontent.com/38676455/71520382-ce835200-28f6-11ea-995e-f5e909c68f08.jpg)
-------------------
>[SaladictTranslate.zip v1.0](https://github.com/crimx/ext-saladict/files/4005179/Saladict.zip)
快捷键调用复制以及呼出沙拉查词窗口。
>

**Reply by funnybones2015** [reply]
> SaladictTranslate v1.8.zip

现在PantherBar更名为Snip.Do了，这个插件貌似不起作用了，无法调起沙拉查词的查词窗口了，重装也不行了。望解决~~

Quicker感觉不如PantherBar简单直观吧 还是更习惯PantherBar

**Reply by systemoutprintlnhelloworld** [reply]
> > SaladictTranslate v1.8.zip
> 
> 现在PantherBar更名为Snip.Do了，这个插件貌似不起作用了，无法调起沙拉查词的查词窗口了，重装也不行了。望解决~~
> 
> Quicker感觉不如PantherBar简单直观吧 还是更习惯PantherBar

没有，我刚刚试过正常使用了，要么这个版本`3.0.41.0`又适配了，要么你可能没在[快捷键菜单](edge://extensions/shortcuts)设置正确的“在独立窗口中搜索剪贴板内容”对应快捷键。

**Reply by hshenmeow** [reply]
用这个动作调用的时候发现一直会出现小键盘锁定的提示（Num Lock 开），不知道是什么问题？

**Reply by BenmiaoZ** [reply]
> 用这个动作调用的时候发现一直会出现小键盘锁定的提示（Num Lock 开），不知道是什么问题？

我的电脑(HP envy x360二合一轻薄本)也有这个问题，经过研究发现了原因并解决了。

原因：
常见于没有小键盘的笔记本，会有num lock功能来提供虚拟小键盘
在上面提供的沙拉查词扩展.ps1脚本中，使用了PowerShell 的 $wshell.SendKeys() 命令来模拟按下快捷键来调用沙拉查词
而出于一些系统层面的原因，这个命令会被识别为{NUMLOCK}功能，因此会触发提示，并且如果有外接键盘的话，真的会关闭你的小键盘功能

解决办法：
使用 .NET Framework 的 SendWait 方法来代替$wshell.SendKeys()
PowerShell 可以访问 .NET Framework 类库，其中 System.Windows.Forms.SendKeys.SendWait() 方法通常比 $wshell.SendKeys() 更加稳定，而且不会依赖 WSH 对象，可能可以避免这个特定的驱动冲突

具体修改代码：
下载压缩包后修改 .ps1 文件，重新压缩修改后缀即可。
```
# Pantherbar 传入的参数
param(
    [string]$TextProcess, # 处理文本判断
    [string]$PLAIN_TEXT     # 纯文本
)

# ... (其他代码保持不变) ...

# 【新增代码】加载 System.Windows.Forms 程序集
Add-Type -AssemblyName System.Windows.Forms 

# 自定义快捷键
# 比如 Alt+l 为 "%l" ; Ctrl+l 为 "^l" ; Shift+l 为 "+l" ; Alt+Shift+l 为 "%+l"
$ShortCut = "%l"

# ... (中间的函数和文本处理部分保持不变) ...

# 【移除】不再需要 $wshell = New-Object -ComObject wscript.shell 这一行

# 定义处理文本模块
function TextModify () {
    $SepText = TextSplit($PLAIN_TEXT)
    $Separator = PunctuationSegment($SepText)
    $JoinText = Textjoin $SepText $Separator
    $JoinText | Set-Clipboard
    
    # 【替换】使用 SendWait 代替 SendKeys
    [System.Windows.Forms.SendKeys]::SendWait($ShortCut)
}

# 定义直接赋值文本模块
function TextCopy () {
    $PLAIN_TEXT | Set-Clipboard
    
    # 【替换】使用 SendWait 代替 SendKeys
    [System.Windows.Forms.SendKeys]::SendWait($ShortCut)
}

if ($TextProcess -eq "process") {
    TextModify
}
elseif ($TextProcess -eq "initial") {
    TextCopy
}
```

@hqyyqh 作者看到的话也可以更新一下脚本

**Reply by hqyyqh** [reply]
感谢反馈。
我下载最新版SnipDo进行测试，尝试提到的SendWait方法，还是有概率出现Num Lock的提示。
不过更主要的问题是，我发现这个插件无法在SnipDo中稳定响应了，点击10次都没有一次响应，头大，所以也没法测试。
![SnipDo_test](https://github.com/user-attachments/assets/125eab64-e7e1-40d2-85db-c6d4abbf4fc4)

你这里使用SnipDo的沙拉查词插件时正常响应吗？
@BenmiaoZ

> > 用这个动作调用的时候发现一直会出现小键盘锁定的提示（Num Lock 开），不知道是什么问题？
> 
> 我的电脑 (HP envy x360 二合一轻薄本) 也有这个问题，经过研究发现了原因并解决了。
> 
> 原因： 常见于没有小键盘的笔记本，会有 num lock 功能来提供虚拟小键盘 在上面提供的沙拉查词扩展.ps1 脚本中，使用了 PowerShell 的 $wshell.SendKeys() 命令来模拟按下快捷键来调用沙拉查词 而出于一些系统层面的原因，这个命令会被识别为 {NUMLOCK} 功能，因此会触发提示，并且如果有外接键盘的话，真的会关闭你的小键盘功能
> 
> 解决办法： 使用 .NET Framework 的 SendWait 方法来代替 $wshell.SendKeys() PowerShell 可以访问 .NET Framework 类库，其中 System.Windows.Forms.SendKeys.SendWait() 方法通常比 $wshell.SendKeys() 更加稳定，而且不会依赖 WSH 对象，可能可以避免这个特定的驱动冲突
> 
> 具体修改代码： 下载压缩包后修改 .ps1 文件，重新压缩修改后缀即可。
> 
> ```
> # Pantherbar 传入的参数
> param(
>     [string]$TextProcess, # 处理文本判断
>     [string]$PLAIN_TEXT     # 纯文本
> )
> 
> # ... (其他代码保持不变) ...
> 
> # 【新增代码】加载 System.Windows.Forms 程序集
> Add-Type -AssemblyName System.Windows.Forms 
> 
> # 自定义快捷键
> # 比如 Alt+l 为 "%l" ; Ctrl+l 为 "^l" ; Shift+l 为 "+l" ; Alt+Shift+l 为 "%+l"
> $ShortCut = "%l"
> 
> # ... (中间的函数和文本处理部分保持不变) ...
> 
> # 【移除】不再需要 $wshell = New-Object -ComObject wscript.shell 这一行
> 
> # 定义处理文本模块
> function TextModify () {
>     $SepText = TextSplit($PLAIN_TEXT)
>     $Separator = PunctuationSegment($SepText)
>     $JoinText = Textjoin $SepText $Separator
>     $JoinText | Set-Clipboard
>     
>     # 【替换】使用 SendWait 代替 SendKeys
>     [System.Windows.Forms.SendKeys]::SendWait($ShortCut)
> }
> 
> # 定义直接赋值文本模块
> function TextCopy () {
>     $PLAIN_TEXT | Set-Clipboard
>     
>     # 【替换】使用 SendWait 代替 SendKeys
>     [System.Windows.Forms.SendKeys]::SendWait($ShortCut)
> }
> 
> if ($TextProcess -eq "process") {
>     TextModify
> }
> elseif ($TextProcess -eq "initial") {
>     TextCopy
> }
> ```
> 
> @hqyyqh 作者看到的话也可以更新一下脚本

**Comment by zhxiyao** [comment]
Awesome！

**Comment by hqyyqh** [comment]
> Awesome！

我今晚把 Quicker 动作文本处理部分都搬了过来，PDF 中识别分段效果很好。除了没法截图翻译，作为桌面划词完全没问题！

**Comment by zhxiyao** [comment]
快捷鍵定義爲 ^(cc) 可用於goldendict 或者歐陸詞典

**Reply by PhDLuffy** [reply]
你好，请问下pantherbar怎么和欧路词典联动啊，你这个快捷键定义为 ^(cc) 是定义到哪个位置呢？

**Comment by leonsong09** [comment]
> 这个也是可以的，使用[文档](http://pantherbar-app.com/create-extension)中的 `keyCombo` 选项或者在 `PowerShell` 脚本中调用快捷键就可以。
> 
> > Pantherbar 大部分场合可用，在少数网页跟软件内不可用（例如 Acrobat）
> 
> ## 效果展示
> ### 划词操作
> ![PantherbarSaladict](https://user-images.githubusercontent.com/38676455/71808606-2f561b80-30a9-11ea-89ca-1ceb128d751f.gif)
> 
> ### 分段识别
> ![Segment](https://user-images.githubusercontent.com/38676455/71808629-3e3cce00-30a9-11ea-8941-fb8914396403.jpg)
> 
> ## PantherBar 沙拉查词扩展下载
> 快捷键设置项【在独立窗口中搜索剪贴板内容】为 Alt+L，全局（沙拉查词快捷键具体设置参考 #493 ）。下载文件后缀`.zip` 改为 `.pbar`，双击安装。
> 
> **自定义快捷键：** 如需自定义调用快捷键，下载压缩包后修改 `.ps1` 文件内的 `$ShortCut` 变量，重新压缩修改后缀即可。
> ![custom shortcuts](https://user-images.githubusercontent.com/38676455/71591644-0fce6880-2b68-11ea-8e74-df6955fc337e.jpg)
> 
> > [SaladictTranslate v1.8.zip](https://github.com/crimx/ext-saladict/files/4022902/SaladictTranslate.v1.8.zip)
> > fix: 删除空行
> > add: 标题判断
> 
> ### 历史版本
> > [SaladictTranslate v1.6.zip](https://github.com/crimx/ext-saladict/files/4010405/SaladictTranslate.v1.6.zip)
> > fix: 误判行末空白字符为分段
> 
> > [SaladictTranslate v1.4.zip](https://github.com/crimx/ext-saladict/files/4007678/SaladictTranslate.v1.4.zip)
> > 
> > 1. 解决 Powershell 输出到剪贴板的编码问题
> > 2. 完善识别分段功能，基本能够还原 PDF 的分段格式
> > 3. 替换 LaTeX 生成 PDF 中 `ﬂ ﬃ ﬁ ﬀ` 字符
> >    **（插件基本完美可用了）**
> >    ![Snipaste_2019-12-28_22-21-53_compress](https://user-images.githubusercontent.com/38676455/71545435-f097d580-29c5-11ea-92e4-2280511ec776.jpg)
> 
> > [SaladictTranslate v1.2.zip](https://github.com/crimx/ext-saladict/files/4006901/SaladictTranslate.v1.2.zip)
> > 增加标点识别分段
> 
> > [SaladictTranslate.zip v1.1](https://github.com/crimx/ext-saladict/files/4006673/SaladictTranslate.zip)
> > 增加是否删除分段选项
> > ![text](https://user-images.githubusercontent.com/38676455/71520382-ce835200-28f6-11ea-995e-f5e909c68f08.jpg)
> 
> > [SaladictTranslate.zip v1.0](https://github.com/crimx/ext-saladict/files/4005179/Saladict.zip)
> > 快捷键调用复制以及呼出沙拉查词窗口。

能问下您用的是什么pdf阅读器吗

**Comment by hqyyqh** [comment]
[PDF-XChange Editor](https://www.tracker-software.com/product/pdf-xchange-editor)

**Comment by Sibyl233** [comment]
请问在其它pdf阅读器里可以使用么？比如在福昕阅读器里划词后正常出现pantherbar，但无法使用沙拉查词

**Comment by hqyyqh** [comment]
> 请问在其它 pdf 阅读器里可以使用么？比如在福昕阅读器里划词后正常出现 pantherbar，但无法使用沙拉查词

检查下是否按照说明配置了沙拉查词的全局快捷键为 Alt+L。

注：
pantherbar限制比较多，我目前没有使用了，扩展也有很长一段时间没有维护，无法确保正常使用。
Windows系统的沙拉查词我更推荐使用 Quicker #493

**Comment by funnybones2015** [comment]
现在PantherBar更名为Snip.Do了，这个插件貌似不起作用了，无法调起沙拉查词的查词窗口了，重装也不行了。望解决~~

Quicker感觉不如PantherBar简单直观吧 还是更习惯PantherBar

**Reply by z0ow** [reply]
我已经给Snip.Do反馈了这个问题
```
[10:10:14:847] RunScript
[10:10:14:856] Path.GetDirectoryName(path) C:\Users\small\AppData\Local\SnipDo\Data\156950082
[10:10:14:856] adding paramter PLAIN_TEXT:Total scan times are dependent on resolution required and the
apparent count rate during the survey. A standard resolution
scan takes up to 2 hours and a high-resolution scan takes up to
8 hours.
[10:10:14:856] adding paramter URLENCODED_ALT_TEXT:Total%20scan%20times%20are%20dependent%20on%20resolution%20required%20and%20the%0D%0Aapparent%20count%20rate%20during%20the%20survey.%20A%20standard%20resolution%0D%0Ascan%20takes%20up%20to%202%20hours%20and%20a%20high-resolution%20scan%20takes%20up%20to%0D%0A8%20hours.
[10:10:14:856] adding paramter URLENCODED_TEXT:Total+scan+times+are+dependent+on+resolution+required+and+the%0d%0aapparent+count+rate+during+the+survey.+A+standard+resolution%0d%0ascan+takes+up+to+2+hours+and+a+high-resolution+scan+takes+up+to%0d%0a8+hours.
[10:10:14:856] adding paramter TextProcess:process
[10:10:14:856] adding paramter HTMLENCODED_TEXT:Total scan times are dependent on resolution required and the
apparent count rate during the survey. A standard resolution
scan takes up to 2 hours and a high-resolution scan takes up to
8 hours.
[10:10:14:871] Result:  .ExitCode: 
在可以调用 OLE 之前，必须将当前线程设置为单线程单元(STA)模式。请确保您的 Main 函数带有 STAThreadAttribute 标记。
[10:10:14:878] Window_Loaded
[10:10:14:883] ShowControlled
[10:10:20:346] AbortHideIfNotOnForm[10:10:20:516]  success
[10:10:22:582] AbortHideIfNotOnForm[10:10:22:631]  success
[10:10:23:593] AbortHideIfNotOnForm
```
这是我得到的Snip.Do的错误，正在找为什么会这样……

**Reply by z0ow** [reply]
> 我已经给Snip.Do反馈了这个问题
> 
> ```
> [10:10:14:847] RunScript
> [10:10:14:856] Path.GetDirectoryName(path) C:\Users\small\AppData\Local\SnipDo\Data\156950082
> [10:10:14:856] adding paramter PLAIN_TEXT:Total scan times are dependent on resolution required and the
> apparent count rate during the survey. A standard resolution
> scan takes up to 2 hours and a high-resolution scan takes up to
> 8 hours.
> [10:10:14:856] adding paramter URLENCODED_ALT_TEXT:Total%20scan%20times%20are%20dependent%20on%20resolution%20required%20and%20the%0D%0Aapparent%20count%20rate%20during%20the%20survey.%20A%20standard%20resolution%0D%0Ascan%20takes%20up%20to%202%20hours%20and%20a%20high-resolution%20scan%20takes%20up%20to%0D%0A8%20hours.
> [10:10:14:856] adding paramter URLENCODED_TEXT:Total+scan+times+are+dependent+on+resolution+required+and+the%0d%0aapparent+count+rate+during+the+survey.+A+standard+resolution%0d%0ascan+takes+up+to+2+hours+and+a+high-resolution+scan+takes+up+to%0d%0a8+hours.
> [10:10:14:856] adding paramter TextProcess:process
> [10:10:14:856] adding paramter HTMLENCODED_TEXT:Total scan times are dependent on resolution required and the
> apparent count rate during the survey. A standard resolution
> scan takes up to 2 hours and a high-resolution scan takes up to
> 8 hours.
> [10:10:14:871] Result:  .ExitCode: 
> 在可以调用 OLE 之前，必须将当前线程设置为单线程单元(STA)模式。请确保您的 Main 函数带有 STAThreadAttribute 标记。
> [10:10:14:878] Window_Loaded
> [10:10:14:883] ShowControlled
> [10:10:20:346] AbortHideIfNotOnForm[10:10:20:516]  success
> [10:10:22:582] AbortHideIfNotOnForm[10:10:22:631]  success
> [10:10:23:593] AbortHideIfNotOnForm
> ```
> 
> 这是我得到的Snip.Do的错误，正在找为什么会这样……

感觉是Snip.D自己的锅，而且它的插件说明页还没搞好https://snipdo-app.com/create-extension ，作者这么匆匆把PantherBar换成SnipD真的翻大车 😞

**Reply by z0ow** [reply]
> > 我已经给Snip.Do反馈了这个问题
> > ```
> > [10:10:14:847] RunScript
> > [10:10:14:856] Path.GetDirectoryName(path) C:\Users\small\AppData\Local\SnipDo\Data\156950082
> > [10:10:14:856] adding paramter PLAIN_TEXT:Total scan times are dependent on resolution required and the
> > apparent count rate during the survey. A standard resolution
> > scan takes up to 2 hours and a high-resolution scan takes up to
> > 8 hours.
> > [10:10:14:856] adding paramter URLENCODED_ALT_TEXT:Total%20scan%20times%20are%20dependent%20on%20resolution%20required%20and%20the%0D%0Aapparent%20count%20rate%20during%20the%20survey.%20A%20standard%20resolution%0D%0Ascan%20takes%20up%20to%202%20hours%20and%20a%20high-resolution%20scan%20takes%20up%20to%0D%0A8%20hours.
> > [10:10:14:856] adding paramter URLENCODED_TEXT:Total+scan+times+are+dependent+on+resolution+required+and+the%0d%0aapparent+count+rate+during+the+survey.+A+standard+resolution%0d%0ascan+takes+up+to+2+hours+and+a+high-resolution+scan+takes+up+to%0d%0a8+hours.
> > [10:10:14:856] adding paramter TextProcess:process
> > [10:10:14:856] adding paramter HTMLENCODED_TEXT:Total scan times are dependent on resolution required and the
> > apparent count rate during the survey. A standard resolution
> > scan takes up to 2 hours and a high-resolution scan takes up to
> > 8 hours.
> > [10:10:14:871] Result:  .ExitCode: 
> > 在可以调用 OLE 之前，必须将当前线程设置为单线程单元(STA)模式。请确保您的 Main 函数带有 STAThreadAttribute 标记。
> > [10:10:14:878] Window_Loaded
> > [10:10:14:883] ShowControlled
> > [10:10:20:346] AbortHideIfNotOnForm[10:10:20:516]  success
> > [10:10:22:582] AbortHideIfNotOnForm[10:10:22:631]  success
> > [10:10:23:593] AbortHideIfNotOnForm
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 这是我得到的Snip.Do的错误，正在找为什么会这样……
> 
> 感觉是Snip.D自己的锅，而且它的插件说明页还没搞好https://snipdo-app.com/create-extension ，作者这么匆匆把PantherBar换成SnipD真的翻大车 😞

经过Debug，应该是powershell代码的Set-Clipboard触发了这个问题，原因很简单……SnipD没有给Main设置STAThread Attribute 😢只能等作者修复了

**Reply by funnybones2015** [reply]
好的，谢谢！！

**Reply by hardsoft888** [reply]
> > 感觉是Snip.D自己的锅，而且它的插件说明页还没搞好https://snipdo-app.com/create-extension ，作者这么匆匆把PantherBar换成SnipD真的翻大车 😞
> 
> 经过Debug，应该是powershell代码的Set-Clipboard触发了这个问题，原因很简单……SnipD没有给Main设置STAThread Attribute 😢只能等作者修复了

说实话，这个SnipD比原来的pantherbar差太多了，最主观的就是相应速度明显变慢了，而且，好多人买的终身版pantherbar，一升级给人家变成了free版。

**Reply by funnybones2015** [reply]
> > > 我已经给Snip.Do反馈了这个问题
> > > ```
> > > [10:10:14:847] RunScript
> > > [10:10:14:856] Path.GetDirectoryName(path) C:\Users\small\AppData\Local\SnipDo\Data\156950082
> > > [10:10:14:856] adding paramter PLAIN_TEXT:Total scan times are dependent on resolution required and the
> > > apparent count rate during the survey. A standard resolution
> > > scan takes up to 2 hours and a high-resolution scan takes up to
> > > 8 hours.
> > > [10:10:14:856] adding paramter URLENCODED_ALT_TEXT:Total%20scan%20times%20are%20dependent%20on%20resolution%20required%20and%20the%0D%0Aapparent%20count%20rate%20during%20the%20survey.%20A%20standard%20resolution%0D%0Ascan%20takes%20up%20to%202%20hours%20and%20a%20high-resolution%20scan%20takes%20up%20to%0D%0A8%20hours.
> > > [10:10:14:856] adding paramter URLENCODED_TEXT:Total+scan+times+are+dependent+on+resolution+required+and+the%0d%0aapparent+count+rate+during+the+survey.+A+standard+resolution%0d%0ascan+takes+up+to+2+hours+and+a+high-resolution+scan+takes+up+to%0d%0a8+hours.
> > > [10:10:14:856] adding paramter TextProcess:process
> > > [10:10:14:856] adding paramter HTMLENCODED_TEXT:Total scan times are dependent on resolution required and the
> > > apparent count rate during the survey. A standard resolution
> > > scan takes up to 2 hours and a high-resolution scan takes up to
> > > 8 hours.
> > > [10:10:14:871] Result:  .ExitCode: 
> > > 在可以调用 OLE 之前，必须将当前线程设置为单线程单元(STA)模式。请确保您的 Main 函数带有 STAThreadAttribute 标记。
> > > [10:10:14:878] Window_Loaded
> > > [10:10:14:883] ShowControlled
> > > [10:10:20:346] AbortHideIfNotOnForm[10:10:20:516]  success
> > > [10:10:22:582] AbortHideIfNotOnForm[10:10:22:631]  success
> > > [10:10:23:593] AbortHideIfNotOnForm
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 这是我得到的Snip.Do的错误，正在找为什么会这样……
> > 
> > 
> > 感觉是Snip.D自己的锅，而且它的插件说明页还没搞好https://snipdo-app.com/create-extension ，作者这么匆匆把PantherBar换成SnipD真的翻大车 😞
> 
> 经过Debug，应该是powershell代码的Set-Clipboard触发了这个问题，原因很简单……SnipD没有给Main设置STAThread Attribute 😢只能等作者修复了

实测目前新版已经修复了这个bug，谢谢您的反馈！！

**Reply by funnybones2015** [reply]
> > > 感觉是Snip.D自己的锅，而且它的插件说明页还没搞好https://snipdo-app.com/create-extension ，作者这么匆匆把PantherBar换成SnipD真的翻大车 😞
> > 
> > 
> > 经过Debug，应该是powershell代码的Set-Clipboard触发了这个问题，原因很简单……SnipD没有给Main设置STAThread Attribute 😢只能等作者修复了
> 
> 说实话，这个SnipD比原来的pantherbar差太多了，最主观的就是相应速度明显变慢了，而且，好多人买的终身版pantherbar，一升级给人家变成了free版。

新版修复了，和以前用起来感觉好像没差别

**Reply by z0ow** [reply]
@hardsoft888 @funnybones2015 兄弟们，我也是受害者，我为了多插件和备份恢复买了终身版没多久，就给我强制变成订阅了，因为受不了这个开发者，目前我正在着手开发一个跟PantherBar（现SnipDo）功能、界面相似的软件，并且打算根据实际情况考虑是否开源，目前引擎部分已经写好了，估计不久将来就可以看到成品了，当然受限于时间问题，可能界面会稍微有点难看，希望大家到时候捧个场

**Reply by hardsoft888** [reply]
> @hardsoft888 @funnybones2015 兄弟们，我也是受害者，我为了多插件和备份恢复买了终身版没多久，就给我强制变成订阅了，因为受不了这个开发者，目前我正在着手开发一个跟PantherBar（现SnipDo）功能、界面相似的软件，并且打算根据实际情况考虑是否开源，目前引擎部分已经写好了，估计不久将来就可以看到成品了，当然受限于时间问题，可能界面会稍微有点难看，希望大家到时候捧个场

期待……

**Reply by hxhc** [reply]
> @hardsoft888 @funnybones2015 兄弟们，我也是受害者，我为了多插件和备份恢复买了终身版没多久，就给我强制变成订阅了，因为受不了这个开发者，目前我正在着手开发一个跟PantherBar（现SnipDo）功能、界面相似的软件，并且打算根据实际情况考虑是否开源，目前引擎部分已经写好了，估计不久将来就可以看到成品了，当然受限于时间问题，可能界面会稍微有点难看，希望大家到时候捧个场

非常期待！

**Comment by hardsoft888** [comment]
谢谢你的提醒，果然搞定了。


强哥也柔情
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "crimx/ext-saladict"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年4月29日(星期六) 中午12:31
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [crimx/ext-saladict] 希望增加到PANTHERBAR  extensions (Discussion #605)





   
SaladictTranslate v1.8.zip
  
现在PantherBar更名为Snip.Do了，这个插件貌似不起作用了，无法调起沙拉查词的查词窗口了，重装也不行了。望解决~~
 
Quicker感觉不如PantherBar简单直观吧 还是更习惯PantherBar
  
没有，我刚刚试过正常使用了，要么这个版本3.0.41.0又适配了，要么你可能没在快捷键菜单设置正确的“在独立窗口中搜索剪贴板内容”对应快捷键。

**Reply by systemoutprintlnhelloworld** [reply]
> 谢谢你的提醒，果然搞定了。 强哥也柔情 ***@***.*** &nbsp;
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "crimx/ext-saladict" ***@***.***&gt;; 发送时间:&nbsp;2023年4月29日(星期六) 中午12:31 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [crimx/ext-saladict] 希望增加到PANTHERBAR extensions (Discussion #605) SaladictTranslate v1.8.zip 现在PantherBar更名为Snip.Do了，这个插件貌似不起作用了，无法调起沙拉查词的查词窗口了，重装也不行了。望解决~~ Quicker感觉不如PantherBar简单直观吧 还是更习惯PantherBar 没有，我刚刚试过正常使用了，要么这个版本3.0.41.0又适配了，要么你可能没在快捷键菜单设置正确的“在独立窗口中搜索剪贴板内容”对应快捷键。 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: ***@***.***&gt;

不客气😊

**Comment by BenmiaoZ** [comment]
我可以在SnipDo中正常稳定使用，用了一晚上没出现无响应的情况。也许检查一下是不是快捷键被占用了？2025年10月12日 13:57，hqyyqh ***@***.***>写道：
感谢反馈。
我下载最新版SnipDo进行测试，尝试提到的SendWait方法，还是有概率出现Num Lock的提示。
不过更主要的问题是，我发现这个插件无法在SnipDo中稳定响应了，点击10次都没有一次响应，头大，所以也没法测试。
SnipDo_test.gif (view on web)
你这里使用SnipDo的沙拉查词插件时正常响应吗？
@BenmiaoZ


用这个动作调用的时候发现一直会出现小键盘锁定的提示（Num Lock 开），不知道是什么问题？

我的电脑 (HP envy x360 二合一轻薄本) 也有这个问题，经过研究发现了原因并解决了。
原因： 常见于没有小键盘的笔记本，会有 num lock 功能来提供虚拟小键盘 在上面提供的沙拉查词扩展.ps1 脚本中，使用了 PowerShell 的 $wshell.SendKeys() 命令来模拟按下快捷键来调用沙拉查词 而出于一些系统层面的原因，这个命令会被识别为 {NUMLOCK} 功能，因此会触发提示，并且如果有外接键盘的话，真的会关闭你的小键盘功能
解决办法： 使用 .NET Framework 的 SendWait 方法来代替 $wshell.SendKeys() PowerShell 可以访问 .NET Framework 类库，其中 System.Windows.Forms.SendKeys.SendWait() 方法通常比 $wshell.SendKeys() 更加稳定，而且不会依赖 WSH 对象，可能可以避免这个特定的驱动冲突
具体修改代码： 下载压缩包后修改 .ps1 文件，重新压缩修改后缀即可。
# Pantherbar 传入的参数
param(
    [string]$TextProcess, # 处理文本判断
    [string]$PLAIN_TEXT     # 纯文本
)

# ... (其他代码保持不变) ...

# 【新增代码】加载 System.Windows.Forms 程序集
Add-Type -AssemblyName System.Windows.Forms 

# 自定义快捷键
# 比如 Alt+l 为 "%l" ; Ctrl+l 为 "^l" ; Shift+l 为 "+l" ; Alt+Shift+l 为 "%+l"
$ShortCut = "%l"

# ... (中间的函数和文本处理部分保持不变) ...

# 【移除】不再需要 $wshell = New-Object -ComObject wscript.shell 这一行

# 定义处理文本模块
function TextModify () {
    $SepText = TextSplit($PLAIN_TEXT)
    $Separator = PunctuationSegment($SepText)
    $JoinText = Textjoin $SepText $Separator
    $JoinText | Set-Clipboard
    
    # 【替换】使用 SendWait 代替 SendKeys
    [System.Windows.Forms.SendKeys]::SendWait($ShortCut)
}

# 定义直接赋值文本模块
function TextCopy () {
    $PLAIN_TEXT | Set-Clipboard
    
    # 【替换】使用 SendWait 代替 SendKeys
    [System.Windows.Forms.SendKeys]::SendWait($ShortCut)
}

if ($TextProcess -eq "process") {
    TextModify
}
elseif ($TextProcess -eq "initial") {
    TextCopy
}

@hqyyqh 作者看到的话也可以更新一下脚本

**Reply by hqyyqh** [reply]
我已经更新下载链接了。

应该是软件冲突或者系统问题， 我在关闭其他软件并且重装软件后可以正常运行。

根据我这里测试的情况
仅仅使用 `[System.Windows.Forms.SendKeys]::SendWait($ShortCut)` 在电脑 `numlock` 打开的情况下还是会弹出 `numlock` 提示， 在发送快捷键之前前面增加一点延时 `Start-Sleep -Milliseconds 100  # 加入100毫秒的延迟` 就可以了。
